internal/controller/profiles/dev/states_dev.go

```diff
@@ -142,6 +151,14 @@ func (e *ensureRunningWorkflowState) Do(ctx context.Context, workflow *operatora
 	return ctrl.Result{RequeueAfter: constants.RequeueAfterIsRunning}, objs, nil
 }
 
+func (e *ensureRunningWorkflowState) ensureServiceMonitor(ctx context.Context, workflow *operatorapi.SonataFlow, pl *operatorapi.SonataFlowPlatform) (client.Object, error) {
+	if monitoring.IsMonitoringEnabled(pl) {
+		serviceMonitor, _, err := e.ensurers.serviceMonitor.Ensure(ctx, workflow)
```

Comment:
We should always fail fast; otherwise, debugging will be hard.

That's a good story, @wmedvede. We can open a follow-up issue for this one. Another option is to query the `/q/metrics` endpoint during the reconciliation and fail there. 

But we can extend the validation story to unpack the available extensions and add them to the `.status.extensions: []`. This way, the operator can behave differently depending on what's available. For instance, if the `.spec.persistence` or `.spec.monitoring` attributes are defined, and there are no extensions, we fail or warn in an event.

Reply:
@wmedvede In this case, the servicemoitor won't cause problems for Prometheus. Prometheus won't find applicable rules to scrape the metrics from the dev profile workflows and of course there is no metrics from such workflows. I think we should keep things simple, and wait for the dev profile image to be ready (and later on we may want to add E2E cases to cover such case).

Reply:
Ok, so the no negative side effects, thanks for the explanation. Yeah, in a follow-up PR we can incorporate these nitpicks.

Reply:
Can you guys summarize this in a new issue?

Reply:
My last comment on this PR guys.
Feel free to open a follow up issue if you believe makes sense.

**case1:**
I have brand new OpenShift Local installation.
I never installed the Prometheus operator
I create a SFP with spec.monitoring.enabled = true
I create a WF in namespace test
After the WF deployment I can see a ServiceMonitor monitor created for it, weird. 
Only explanation is that OpenShift Local already has the operator installed the Prometheus resources ouf-of-the-box

In customer OpenShfift cluster installations different thant OpenShift Local installations IDK what will happen. Probably the operator is always there.

**case2:**
I install the Prometheus operator in namespace test1
I create a SFP, with spec.monitoring.enabled = true
I create, a WF, etc,
I can see the ServiceMonitior, etc.

I create a namespace test2, I don't install the Prometheus operator in that namespace.
I create a SFP, with spec.monitoring.enabled = true
I create, a WF, etc,
I can see that a ServiceMonitior, etc.

Sort of corolary o case1